lib/Search/ConversationSearch.php

```diff
@@ -113,36 +117,21 @@ public function search(IUser $user, ISearchQuery $query): SearchResult {
 				}
 			}
 
-			$icon = '';
-			$iconClass = '';
-			if ($room->getType() === Room::TYPE_ONE_TO_ONE) {
-				$users = json_decode($room->getName(), true);
-				foreach ($users as $participantId) {
-					if ($participantId !== $user->getUID()) {
-						$icon = $this->url->linkToRouteAbsolute('core.avatar.getAvatar', [
-							'userId' => $participantId,
-							'size' => 512,
-						]);
-					}
-				}
-			} elseif ($room->getObjectType() === 'file') {
-				$iconClass = 'conversation-icon icon-file-white';
-			} elseif ($room->getObjectType() === 'share:password') {
-				$iconClass = 'conversation-icon icon-password-white';
-			} elseif ($room->getObjectType() === 'emails') {
-				$iconClass = 'conversation-icon icon-mail-white';
-			} elseif ($room->getType() === Room::TYPE_PUBLIC) {
-				$iconClass = 'conversation-icon icon-public-white';
-			} else {
-				$iconClass = 'conversation-icon icon-contacts-white';
+			$arguments = [
+				'token' => $room->getToken(),
+				'apiVersion' => 'v1',
+			];
+			if ($avatar = $room->getAvatar()) {
+				$arguments['v'] = $avatar;
 			}
+			$icon = $this->url->linkToOCSRouteAbsolute('spreed.Avatar.getAvatar', $arguments);
```

Comment:
Why does this search here have so much special handling.

Calling `$this->avatarService->getAvatarUrl($room)` like in dashboard, activity and notifications should be good enough?!